assets/js/modules/analytics-4/components/widgets/TopTrafficSourceWidget.js

```diff
@@ -21,27 +21,123 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
+import MetricTileText from '../../../../components/KeyMetrics/MetricTileText';
 import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
 
-const { useSelect } = Data;
+const { useSelect, useInViewSelect } = Data;
 
 export default function TopTrafficSourceWidget( { Widget, WidgetNull } ) {
 	const keyMetricsWidgetHidden = useSelect( ( select ) =>
 		select( CORE_USER ).isKeyMetricsWidgetHidden()
 	);
```

Comment:
We need to use the new `whenKeyMetricsWidgetVisible` HOC as we do it in NewVisitors and LoyalVisitors widgets:  https://github.com/google/site-kit-wp/blob/8fe4cfb0510912af79ff7604b1bd11f717cc5703/assets/js/modules/analytics-4/components/widgets/LoyalVisitorsWidget.js#L116

Once you apply that HOC to this component, we will need to remove `keyMetricsWidgetHidden` usage in the component.

Reply:
@jimmymadon, I left a comment in that ticket and moved it back to IB. Please, use HOC for you widget.